metadata-service/factories/src/main/java/com/linkedin/gms/factory/timeline/EntityChangeEventGeneratorRegistryFactory.java

```diff
@@ -0,0 +1,65 @@
+import javax.annotation.Nonnull;
+import javax.inject.Singleton;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+
+import static com.linkedin.metadata.Constants.*;
+
+
+@Configuration
+public class EntityChangeEventGeneratorRegistryFactory {
+
+  @Bean(name = "entityChangeEventGeneratorRegistry")
+  @Singleton
+  @Nonnull
+  protected com.linkedin.metadata.timeline.eventgenerator.EntityChangeEventGeneratorRegistry entityChangeEventGeneratorRegistry() {
+    final com.linkedin.metadata.timeline.eventgenerator.EntityChangeEventGeneratorRegistry registry =
+        new com.linkedin.metadata.timeline.eventgenerator.EntityChangeEventGeneratorRegistry();
+    registry.register(SCHEMA_METADATA_ASPECT_NAME, new SchemaMetadataChangeEventGenerator());
+    registry.register(EDITABLE_SCHEMA_METADATA_ASPECT_NAME, new EditableSchemaMetadataChangeEventGenerator());
+    registry.register(GLOBAL_TAGS_ASPECT_NAME, new GlobalTagsChangeEventGenerator());
+    registry.register(GLOSSARY_TERMS_ASPECT_NAME, new GlossaryTermsChangeEventGenerator());
+    registry.register(OWNERSHIP_ASPECT_NAME, new OwnershipChangeEventGenerator());
+    registry.register(INSTITUTIONAL_MEMORY_ASPECT_NAME, new InstitutionalMemoryChangeEventGenerator());
+    registry.register(DATASET_PROPERTIES_ASPECT_NAME, new DatasetPropertiesChangeEventGenerator());
+    registry.register(DOMAINS_ASPECT_NAME, new SingleDomainChangeEventGenerator());
+    registry.register(DATASET_PROPERTIES_ASPECT_NAME, new DatasetPropertiesChangeEventGenerator());
+    registry.register(EDITABLE_DATASET_PROPERTIES_ASPECT_NAME, new EditableDatasetPropertiesChangeEventGenerator());
+
+    // Entity Lifecycle Differs
+    registry.register(DATASET_KEY_ASPECT_NAME, new EntityKeyChangeEventGenerator<>());
+    registry.register(CONTAINER_KEY_ASPECT_NAME, new EntityKeyChangeEventGenerator<>());
+    registry.register(CHART_KEY_ASPECT_NAME, new EntityKeyChangeEventGenerator<>());
+    registry.register(DASHBOARD_KEY_ASPECT_NAME, new EntityKeyChangeEventGenerator<>());
+    registry.register(DATA_FLOW_KEY_ASPECT_NAME, new EntityKeyChangeEventGenerator<>());
+    registry.register(DATA_JOB_KEY_ASPECT_NAME, new EntityKeyChangeEventGenerator<>());
+    registry.register(DOMAIN_KEY_ASPECT_NAME, new EntityKeyChangeEventGenerator<>());
+    registry.register(TAG_KEY_ASPECT_NAME, new EntityKeyChangeEventGenerator<>());
+    registry.register(GLOSSARY_TERM_KEY_ASPECT_NAME, new EntityKeyChangeEventGenerator<>());
+    registry.register(CORP_GROUP_KEY_ASPECT_NAME, new EntityKeyChangeEventGenerator<>());
+    registry.register(STATUS_ASPECT_NAME, new StatusChangeEventGenerator());
+    registry.register(DEPRECATION_ASPECT_NAME, new DeprecationChangeEventGenerator());
+
+    // Assertion differ
+    registry.register(ASSERTION_RUN_EVENT_ASPECT_NAME, new AssertionRunEventChangeEventGenerator());
```

Comment:
Do we need to register the Data Job one as well?

Reply:
Good catch, added!